SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialGameInstance.cpp

```diff
@@ -134,17 +134,24 @@ void USpatialGameInstance::TryConnectToSpatial()
 		// Native Unreal creates a NetDriver and attempts to automatically connect if a Host is specified as the first commandline argument.
 		// Since the SpatialOS Launcher does not specify this, we need to check for a locator loginToken to allow automatic connection to provide parity with native.
 
-		// Initialize a locator configuration which will parse command line arguments.
-		FLocatorConfig LocatorConfig;
-		if (LocatorConfig.TryLoadCommandLineArgs())
+		if (!GetDefault<USpatialGDKSettings>()->GetPreventClientCloudDeploymentAutoConnect())
```

Comment:
What's the purpose of creating the connection manager if we're not going to connect?

Reply:
True. This PR is kind of a revert of what used to exist.

Sahil also tested:

void USpatialGameInstance::StartGameInstance()
{
	if (!GetDefault<USpatialGDKSettings>()->GetPreventClientCloudDeploymentAutoConnect(true))
	{
		TryConnectToSpatial();
	}
	Super::StartGameInstance();
}

Which he said was working for him. Perhaps we want to go with that instead.

Reply:
Sahils suggestion a test.

Reply:
Yeah I think that works, although I'd make `TryConnectoToSpatial` a private function then.